xmhf/src/xmhf-core/xmhf-runtime/xmhf-eventhub/arch/x86/vmx/peh-x86vmx-main.c

```diff
@@ -164,8 +164,10 @@ static void _vmx_handle_intercept_cpuid(VCPU *vcpu, struct regs *r){
 		      :"a"(r->eax), "c" (r->ecx));
 
 		if (old_eax == 0x1U) {
+#ifndef __NESTED_VIRTUALIZATION__
```

Comment:
In this function, not line 167:
(1) Need definition in comments for APP_CPUID_CHAIN and APP_CPUID_SKIP
(2) Replace the asm volatile in line 162 with <cpuid> in _processor.h

Reply:
(3) For this PR I don't think there are huge changes made in the hypapp.
1. I am assuming that you are running the configuration `hardware -> XMHF (hypapp=GreenBox) -> Ubuntu`
2. Does TrustVisor boot correctly? Or is this a problem only in GreenBox?
3. Does the problem still appear if compiled with `--disable-nested-virtualization`?
    * If it still appears, then we have less to debug
    * If it no longer appears, then we need to check whether Ubuntu is accessing hypervisor features
4. You can compile with `--enable-debug-event-logger`, which can show you whether XMHF is still active

Reply:
> (1) Added comments (in both `xmhf64` branch and this branch). However ideally should refer to comments in `xmhf-app.h` (2) Unfortunately the macro in `_processor.h` does not handle ECX as argument. For example, CPUID(EAX=7, ECX=0) and CPUID(EAX=7, ECX=1) are different.

Add a new cpuid macro in _processor.h would be better

Reply:
> (3) For this PR I don't think there are huge changes made in the hypapp.
> 
> 1. I am assuming that you are running the configuration `hardware -> XMHF (hypapp=GreenBox) -> Ubuntu`
> 2. Does TrustVisor boot correctly? Or is this a problem only in GreenBox?
> 3. Does the problem still appear if compiled with `--disable-nested-virtualization`?
>    
>    * If it still appears, then we have less to debug
>    * If it no longer appears, then we need to check whether Ubuntu is accessing hypervisor features
> 4. You can compile with `--enable-debug-event-logger`, which can show you whether XMHF is still active

(1) Right
(2) TrustVisor works with "./configure --with-approot=hypapps/trustvisor --enable-drt=no --with-target-subarch=amd64". GreenBox does not work with "./configure --with-approot=../secbase/hypapps/whitevisor --enable-drt=no --with-target-subarch=amd64 --enable-quiesce-in-guest-mem-pio-traps=no --enable-allow-hypapp-disable-igfx-iommu=yes". Any ideas?

Reply:
(1) Added new macro `cpuid_raw`
(3) I do not have a lot of ideas. Since nested-virtualization is disabled, there should be something not directly related to nested virtualization that breaks greenbox. i.e. something outside of `#ifdef __NESTED_VIRTUALIZATION__`. I think we need more bisecting / debugging.

Reply:
To make sure, does `xmhf64` branch work well?